### DISCUSSION TITLE
Redirect non-www to www doesn't work

### DISCUSSION THREAD

**Discussion by xletter-dev** [vercel/community#3368] [Help]
### Summary

I've deployed my project using vercel to [www.xletter.io](https://www.xletter.io/)
and I also set xletter.io to be redirected to www.

[www.xletter.io](https://www.xletter.io/) works well, but redirection from xletter.io doesn't work.
Actually xletter.io is not reachable at all.

What is the problem? And how to fix it?

### Example

https://xletter.io

### Steps to Reproduce

.

**Comment by swarnava** [comment]
Hi,

I can see that the website is being redireced successfully  from xletter.io to www.xletter.io. Are you still seeing this problem?

**Reply by xletter-dev** [reply]
Yes I'm still having the problem. It works sometime, but now it doesn't work again

**Comment by montagao** [comment]
I also get this problem 
<img width="862" alt="Screenshot 2023-07-27 at 2 20 30 PM" src="https://github.com/vercel/community/assets/9814181/2ab3c32b-5137-4c94-bb29-9f6a8c781112">
For my site at https://translate.mom. 

Using google chrome on macOS Ventura 13.5. 

xletter.io also won't load for me.

**Reply by amyegan** [reply]
It's working for me now. Are you still experiencing this problem?

**Comment by xletter-dev** [comment]
> It's working for me now. Are you still experiencing this problem?

Yes I'm still having the problem. It works sometime, but now it doesn't work again

**Reply by montagao** [reply]
same here

**Comment by jaiten-made** [comment]
+1

**Comment by nwazuo** [comment]
+1 for this problem also. All my client domains face the same issue and I might just have to move them all away from Vercel.

**Reply by versecafe** [reply]
@nwazuo would you be able to share your links & your domain configuration?

**Reply by nwazuo** [reply]
gtkate.com

A record is configured based on Vercel's recommendation - 76.76.21.21 but I suspect this IP address might be blacklisted by my ISP. 
Can you verify with `curl -v https://gtkate.com` if this goes through for you?

**Reply by versecafe** [reply]
```
*   Trying 76.76.21.21:443...
* Connected to gtkate.com (76.76.21.21) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=gtkate.com
*  start date: Apr 13 10:18:22 2024 GMT
*  expire date: Jul 12 10:18:21 2024 GMT
*  subjectAltName: host "gtkate.com" matched cert's "gtkate.com"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://gtkate.com/
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: gtkate.com]
* [HTTP/2] [1] [:path: /]
* [HTTP/2] [1] [user-agent: curl/8.4.0]
* [HTTP/2] [1] [accept: */*]
> GET / HTTP/2
> Host: gtkate.com
> User-Agent: curl/8.4.0
> Accept: */*
>
< HTTP/2 308
< cache-control: public, max-age=0, must-revalidate
< content-type: text/plain
< date: Thu, 02 May 2024 16:28:21 GMT
< location: https://www.gtkate.com/
< refresh: 0;url=https://www.gtkate.com/
< server: Vercel
< strict-transport-security: max-age=63072000
< x-vercel-id: iad1::c7dhz-1714667301478-1941148a3640
<
Redirecting...
* Connection #0 to host gtkate.com left intact
```

**Reply by FesoQue** [reply]
> gtkate.com
> 
> A record is configured based on Vercel's recommendation - 76.76.21.21 but I suspect this IP address might be blacklisted by my ISP. Can you verify with `curl -v https://gtkate.com` if this goes through for you?

Hello @nwazuo, were you able to solve the issue because i am currently facing the same problem. If yes, what approach did you take??

**Reply by nwazuo** [reply]
Hey @FesoQue I was. I wrote up something here in a related issue - https://github.com/orgs/vercel/discussions/3630#discussioncomment-9482248

**Reply by FesoQue** [reply]
> Hey @FesoQue I was. I wrote up something here in a related issue - https://github.com/orgs/vercel/discussions/3630#discussioncomment-9482248

Thanks brother @nwazuo, went through the writeup and I want to ask some noobs questions.  I hope you do not mind.
using cloudflare DNS means I am moving away from hosting on vercel, right?
If i am wrong, then how does cloudflare comes into the picture... Currently, I am using the nameserver vercel gave me on namecheap.

Thank you!!!

**Reply by nwazuo** [reply]
Fellow noob here😁 @FesoQue. 

No, you're not moving away from hosting on Vercel. Your Nameserver(which is hosted by a DNS provider) is the server that resolves your domain name(and other things connected to it like email) to an IP address. So, you can use a different DNS provider and still keep your host. Namecheap provides DNS services, so does Vercel and Cloudflare.

You're mostly just changing your nameserver records on your namecheap panel, and then configuring all DNS-related related stuff on Cloudflare instead (which supports Flatenned CNAME to help bypass the issue.

**Reply by FesoQue** [reply]
> Fellow noob here😁 @FesoQue.
> 
> No, you're not moving away from hosting on Vercel. Your Nameserver(which is hosted by a DNS provider) is the server that resolves your domain name(and other things connected to it like email) to an IP address. So, you can use a different DNS provider and still keep your host. Namecheap provides DNS services, so does Vercel and Cloudflare.
> 
> You're mostly just changing your nameserver records on your namecheap panel, and then configuring all DNS-related related stuff on Cloudflare instead (which supports Flatenned CNAME to help bypass the issue.

My bro @nwazuo, should the existing DNS configuration I have on namecheap be deleted after configuring all DNS-related related stuff on Cloudflare??

**Reply by nwazuo** [reply]
Doesn't really matter and I wouldn't advise to delete. In your Namecheap dashboard, you need to switch from Basic DNS to Custom DNS. When you switch to Custom DNS, your DNS settings on Namecheap no longer apply.

**Reply by FesoQue** [reply]
I

> Doesn't really matter and I wouldn't advise to delete. In your Namecheap dashboard, you need to switch from Basic DNS to Custom DNS. When you switch to Custom DNS, your DNS settings on Namecheap no longer apply.

Thanks chief, got it working now!
